Obtain X-ray lines from incoming arguments rather than metadata

dmlundeby opened the pull request:
### Description of the change
Obtain X-ray lines from the intensity signals, rather than from metadata.

### Progress of the PR
- [x] Change implemented
- [x] add tests
- [x] ready for review

### Minimal example of the bug fix or the new feature
```python
>>> import hyperspy.api as hs
>>> s = hs.load(some_eds_signal, signal_type='EDS_TEM')
>>> s.set_elements(['Al', 'Zn'])
>>> s.add_lines(only_one=False)
>>> s.quantification(s.get_lines_intensity(xray_lines=['Al_Ka', 'Zn_Ka'], method='CL', factors=[1, 1])
```
This does no longer raise an exception.

My first attempt on bugfix here, hope it works :-)

Should fix #2061

dmlundeby committed:
Obtain X-ray lines from incoming arguments rather than metadata

k8macarthur commented:
Good spot! This fix looks good to me. It needs a test added. A simple test in the quantification section of test_eds_tem.py should suffice. Check that the x-ray lines used for quantification are those in the intensities list.

dmlundeby committed:
Add test for element order in quantification

Tests the order of elements, when X-ray lines are not in the internal list.
Both the name of the signals and their intensity is tested.

ericpre commented:
Thanks @dmlundeby.
Your name on github will be used to make the list of contributor for the next release, so you may want to also add your last name in order to have your full name on the list. This is also used by zenodo when making a DOI.
